.github/workflows/build_and_test.yml

```diff
@@ -108,10 +151,11 @@ jobs:
         run: poetry run mypy .
 
   check-codestyle-rust:
-    needs: [ check-permissions ]
+    if: always() && contains(fromJSON('["success", "skipped"]'), needs.build-buildtools-image.result)
```

Comment:
I thought we didn't need this logic anymore, am I missing something?

Reply:
yep i have already removed had forgot to push it